src/sql/workbench/services/notebook/browser/sql/sqlSessionManager.ts

```diff
@@ -579,24 +579,32 @@ export class SQLFuture extends Disposable implements FutureInternal {
 		};
 	}
 
-	private convertToHtmlTable(columns: IDbColumn[], d: QueryExecuteSubsetResult): string {
-		let htmlString = '<table>';
+	private convertToHtmlTable(columns: IDbColumn[], d: QueryExecuteSubsetResult): string[] {
+		// Adding 3 for <table>, column title rows, </table>
+		let htmlStringArr: string[] = new Array(d.resultSubset.rowCount + 3);
+		htmlStringArr[0] = '<table>';
 		if (columns.length > 0) {
-			htmlString += '<tr>';
+			let columnHeaders = '';
+			columnHeaders += '<tr>';
 			for (let column of columns) {
-				htmlString += '<th>' + escape(column.columnName) + '</th>';
+				columnHeaders += '<th>' + escape(column.columnName) + '</th>';
 			}
-			htmlString += '</tr>';
+			columnHeaders += '</tr>';
+			htmlStringArr[1] = columnHeaders;
 		}
+		let i = 2;
 		for (const row of d.resultSubset.rows) {
-			htmlString += '<tr>';
+			let rowData = '';
+			rowData += '<tr>';
 			for (let i = 0; i < columns.length; i++) {
```

Comment:
Would be good to use a different var name here since you're already using i above, less confusing which is which

Reply:
Also, would this overwrite the column information? The rowData gets set to index 0, since i is reinitialized.